hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -8,14 +8,290 @@ title: "OTP Controller Technical Specification"
+
+### Life Cycle Partition
+
+The life cycle partition cannot be locked and will therefore not contain a stored digest.
+
+## Secret vs Non-Secret Partitions
+
+Non-secret OTP partitions hold data that can be public; or data that has no impact on security.
+For example, the current value of lock bits, life cycle state or clock calibration value.
+These values are stored in OTP as plaintext.
+
+Secret partitions contain data that are critical to security, for example flash scrambling keys, device root secret and unlock tokens.
+These values are stored scrambled in OTP, and are descrambled upon read.
+The currently employed cipher is PRESENT, as it lends itself well to iterative decomposition, and it is a proven lightweight block cipher (see also [PRESENT Scrambling Primitive]({{< relref "hw/ip/prim/doc/prim_present.md" >}}).
+The usage of a block cipher however implies that the secret partitions can only be written in 64bit chunks.
+
+Further, the contents of a particular secret partition are not readable by software once locked (other than the digest which must be always readable); while non-secret partitions are always readable unless read accessibility is explicitly removed by software.
+
+Unfortunately, secret partitions must utilize a global netlist key for the scrambling operation, as there is no other non-volatile storage to store a unique key.
+
+
+## Partition Checks
+
+### Integrity
+
+Once the appropriate partitions have been locked, the hardware integrity checker employs two integrity checks to verifies the content of the volatile buffer registers:
+
+1. All buffered partitions have additional byte parity protection that is concurrently monitored.
+2. The digest of the partition is recomputed at semi-random intervals and compared to the digest stored alongside the partition.
+
+The purpose of this check is NOT to check between the storage flops and the OTP, but whether the buffer register contents remain consistent with the calculated digest.
+This verification is primarily concerned with whether the storage flops have experienced fault attacks.
+This check applies to only the HW_CFG and SECRET* partitions.
+If a failure is encountered, the OTP controller will send out an `otp_integrity_mismatch` alert and reset all of its hardware outputs to their defaults (**TBD do we need an LFSR clearing mechanism here for the secret partitions?**)
+
+### Storage Consistency
+
+This verification ensures the value stored in the buffer registers remain consistent with those in the OTP.
+This process re-reads the OTP at semi-random intervals and confirms the value read is the same as the value stored.
+Note, given there are integrity checks in parallel, it is not necessary for some partitions to check ALL read contents for consistency.
+If there is an integrity digest, only the digest needs to be read; otherwise, all values must be read.
+
+
+This check applies to LIFE_CYCLE, HW_CFG and SECRET* partitions.
```

Comment:
do we need this check for `HW_CFG` and `SECRET`? Those will have digests right?

Reply:
yes that is right. that is up for us to decide. it would certainly simplify the partition controllers a bit if we did that.

also, everything that is buffered (including LIFE_CYCLE) has Byte parity.
so, given that LIFE_CYCLE is redundantly encoded, you could argue that LIFE_CYCLE doesn't need this consistency check, neither...